src/main/java/org/jboss/pnc/deliverablesanalyzer/Finder.java

```diff
@@ -216,7 +236,10 @@ private FinderResult find(
         analyzer.setListener(distributionAnalyzerListener);
 
         Future<Map<ChecksumType, MultiValuedMap<String, LocalFile>>> futureChecksum = pool.submit(analyzer);
-        result = findBuilds(id, url, analyzer, futureChecksum, buildFinderListener);
+        if (allTasks != null) {
```

Comment:
Are these checks necessary, or shouldn't it be up to the caller not to pass `null`? Because in fact it would be legal to pass `Collections.emptyList()` and you're going to end up with a problem as well.

Reply:
This check will be removed in the next commit push. Thanks for the suggestion.